pkg/controller/controller_utils.go

```diff
@@ -54,6 +54,13 @@ var (
 	KeyFunc = framework.DeletionHandlingMetaNamespaceKeyFunc
 )
 
+type ResyncPeriodFunc func() time.Duration
```

Comment:
It can call the function more than once, because it can have more than one Informer (one informer per resource). I wanted those also to not collide (e.g. daemon controller is using this function for both pods and nodes).